kolibri/core/assets/src/content-renderer.vue

```diff
@@ -0,0 +1,71 @@
+<template>
+
+  <div>
+    <h3>
+      {{ title }}
+    </h3>
+    <div v-el:container></div>
+  </div>
+
+</template>
+
+
+<script>
+
+  module.exports = {
+    props: {
+      channelId: String,
+      contentId: String,
+      kind: String,
+      extension: String,
+      title: String,
+      contentData: {
+        type: Object,
+        default: () => ({ url: '#' }),
+      },
+
+    },
+    computed: {
+      contentType() {
+        return `${this.kind}/${this.extension}`;
+      },
+    },
+    created() {
+      this.Kolibri.once(`component_render:${this.contentType}`, this.setRendererComponent);
+      this.Kolibri.emit(`content_render:${this.contentType}`);
+    },
+    ready() {
+      this.ready = true;
+      this.renderContent();
+    },
+    data: () => ({
+      currentViewClass: null,
+    }),
+    methods: {
+      setRendererComponent(component) {
+        this.currentViewClass = component;
+        if (this.ready && !this.contentView) {
```

Comment:
If the component has already readied itself, but no contentView is available, render the content view.